### PR TITLE
Set up repository

### PR DESCRIPTION
**Pull request by emilyliu7321**
**Description:**

- Add gitignore, codeowners, and PR template
- Add unit test workflow
- Set up branch protections and permissions
- Updated Github labels

**Test steps:**
1. Unit tests pass

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -0,0 +1,23 @@
+name: test
+on: [push, pull_request]
+jobs:
+  test:
+    name: Run unit tests
+    runs-on: ubuntu-latest
+    timeout-minutes: 5
+    strategy:
+      matrix:
+        python-version: [3.8]
```
Comment (chriszs) [inline .github/workflows/test.yml]: Should this be 3.7, since that's the Python data-importer uses?

**Review by chriszs — APPROVED** [review]

File: .github/workflows/test.yml (hunk 1 above)
Comment (emilyliu7321) [reply .github/workflows/test.yml]: I think the data importer version should get bumped up instead!

Comment (emilyliu7321) [reply .github/workflows/test.yml:10]: I think the data importer version should get bumped up instead!

Comment (chriszs) [reply .github/workflows/test.yml]: Would it make any sense to matrix both versions for now, e.g. 
```suggestion
        python-version: [3.8, 3.7]
```

Comment (emilyliu7321) [reply .github/workflows/test.yml]: sounds good!!

**Pull request merged**